[GOOGLE DAYS] Ninjava Google Days series (1/6) (#282)
iodigital-com/io-technology: data/blog/google-days/io-digitals-java-team-adventure-in-container-orchestration.md

---
title: 'Navigating the Kubernetes Sea: iO Digital Java Team Adventure in Container Orchestration'

> date: '2025-01-31'

Reviewer: Once you're done to publish, can you update this date?

Author: Yes, but first will rewrite the other article or even integrate it into this not sure

tags: ['kubernetes', 'devops', 'docker']
images:
  ['/articles/io-digitals-java-team-adventure-in-container-orchestration/kubenetes-workshop.png']

> summary: Recently, our Java team embarked on an exciting journey into the world of Kubernetes, exploring its capabilities and potential for enhancing our development and deployment processes. This "Google Day" initiative allowed our team members to dive deep into Kubernetes, experimenting with advanced deployment techniques like A/B testing and Canary releases. Here's a recap of our adventures and discoveries.

Reviewer: ```suggestion
summary: Recently, our Java team looked into Kubernetes, exploring its capabilities and potential for enhancing our development and deployment processes. This "Google Day" initiative allowed our team members to dive deep into Kubernetes, experimenting with advanced deployment techniques like A/B testing and Canary releases. Here's a recap of our adventures and discoveries.
```

authors: ['arno-koehler']
theme: 'beige'
serie: 'google-days'

> ---

Reviewer: Could you add some tags?

![kubenetes-workshop.png](/articles/io-digitals-java-team-adventure-in-container-orchestration/kubenetes-workshop.png)

> # Navigating the Kubernetes Sea: iO Digital's Java Team's Adventure in Container Orchestration

Reviewer: You can remove this as the title is already at the top of the page through the `title` field in the frontmatter markdown

> At iO Digital, we encourage curiosity and challenge our developers to keep learning and growing. That is why recently, our Java team sailed onto a journey into the world of Kubernetes, exploring its capabilities and potential for enhancing our development and deployment processes. On our so-called "Google Day" we tried to learn about deployment techniques like A/B testing and Canary releases. Here's a recap of our adventure and discoveries we did that day.

Reviewer: Same goes for this intro as it is already at the top of the page through the `summary` field

---
title: 'Navigating the Kubernetes Sea: iO Digital Java Team Adventure in Container Orchestration'
date: '2025-01-31'
tags: ['kubernetes', 'devops', 'docker']
images:
  ['/articles/io-digitals-java-team-adventure-in-container-orchestration/kubenetes-workshop.png']
summary: Recently, our Java team embarked on an exciting journey into the world of Kubernetes, exploring its capabilities and potential for enhancing our development and deployment processes. This "Google Day" initiative allowed our team members to dive deep into Kubernetes, experimenting with advanced deployment techniques like A/B testing and Canary releases. Here's a recap of our adventures and discoveries.
authors: ['arno-koehler']
theme: 'beige'
serie: 'google-days'
---

> ![kubenetes-workshop.png](/articles/io-digitals-java-team-adventure-in-container-orchestration/kubenetes-workshop.png)

Reviewer: This can be removed :)

For us as developers Kubernetes is a way to declare infra as Yaml, that will be translated into:

- **Clusters:** A set of worker machines (nodes) that run containerized applications.
- **Pods:** The smallest deployable units, usually containing one container or a group of tightly coupled containers.
- **Deployments:** Declarative management of application state and rolling updates.
- **Services:** An abstraction of a (set of) pods and a policy to access them.
- **ReplicaSets:** Ensures the desired number of pod replicas are running (though often managed indirectly through Deployments).

Kubernetes offers a way of doing declarative configuration, with great portability and scalability.
All major cloud providers offer managed Kubernetes services.
Google has GKE (Google Kubernetes Engine), Azure has AKS (Azure Kubernetes Service) and Amazone has EKS (Amazon Elastic Kubernetes Service), making it easier for enterprises to adopt and maintain Kubernetes clusters.
The fact that Kubernetes is backed by the Cloud Native Computing Foundation (CNCF) and major tech companies, makes it a great choice for configuring your containers.

> # The Power of Basic Kubernetes: Blue-Green Releases Demystified

Reviewer: This sudden title seems a bit odd. Should it be an h2?
